### PR TITLE
ui: new record landing page (webpack built)

### PR DESCRIPTION
**Pull request by ppanero**
Changes:

- Moves all configuration and code related to records back from `invenio-app-rdm`
- Implements a new UI for the record landing page
- Builds it using webpack (namespaced)
- Adds `badges` extra to install `CairoSVG`
- Adds a *Coming Soon* page where all non-yet-implemented features should point to:
Selection_111

How the result should look like:
Selection_110

And the files part, not collapsed:

Selection_109

Opens new issues (todos):

- Citation box (https://github.com/inveniosoftware/invenio-rdm-records/issues/56)
- Metrics (https://github.com/inveniosoftware/invenio-rdm-records/issues/57)
- Exports (https://github.com/inveniosoftware/invenio-rdm-records/issues/58)
- Versioning (https://github.com/inveniosoftware/invenio-rdm-records/issues/59)

Note: That many components are pure HTML to showcase how it would be in the end.

Will make tests pass after fixing all possible comments :)

### CODE REVIEW

File: invenio_rdm_records/cli.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def create_fake_record():
             "subtype": "photo"
         },
         "identifiers": [{
-            "identifier": "10.5281/zenodo.9999999",
+            "identifier": "99.9999/rdm.9999999",
```
Comment (lnielsen) [inline invenio_rdm_records/cli.py]: DOI's always start with ``10.`` so if at any point we need to validate them, it's better to have it start with ``10.``

File: invenio_rdm_records/theme/__init__.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#
+# Copyright (C) 2020 CERN.
+# Copyright (C) 2020 Northwestern University.
+#
+# Invenio RDM Records is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio RDM Records."""
+
+from __future__ import absolute_import, print_function
```
Comment (lnielsen) [inline invenio_rdm_records/theme/__init__.py]: We can skip these guys now, as we're no longer supporting Python 2

File: invenio_rdm_records/theme/templates/invenio_rdm_records/iiif_preview.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+{%- extends config.PREVIEWER_ABSTRACT_TEMPLATE %}
+
+{%- block head %}
+{% if config.PREVIEWER_ASSETS_USE_WEBPACK %}
+    {{ webpack['invenio-rdm-records-theme.css'] }}
+    {% for css_bundle in css_bundles %}
+    {{ webpack[css_bundle] }}
+    {% endfor %}
+{% else %}
+    {%- for bdl in css_bundles %}
```
Comment (lnielsen) [inline invenio_rdm_records/theme/templates/invenio_rdm_records/iiif_preview.html]: We only need to support webpack, so this can be removed

File: invenio_rdm_records/theme/templates/invenio_rdm_records/macros/files.html [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{#
+    Copyright (C) 2020 CERN.
+    Copyright (C) 2020 Northwestern University.
+  
+    Invenio App RDM is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+#}
+
+{%- macro file_list(files) %}
```
Comment (lnielsen) [inline invenio_rdm_records/theme/templates/invenio_rdm_records/macros/files.html]: There should already be a marco available - see cookiecutter-invenio-instance for where it is.

File: invenio_rdm_records/config.py [hunk 5]
```diff
@@ -120,6 +120,87 @@ def _(x):
...
+    'recid_previewer': {
+        'pid_type': 'recid',
+        'record_class': 'invenio_records_files.api:Record',
+        'route': '/records/<pid_value>/preview/<path:filename>',
+        'view_imp': 'invenio_previewer.views.preview',
+    },
+}
+
+#: Allow list of contributor types.
+RECORD_CONTRIBUTOR_TYPES = [
```
Comment (ntarocco) [inline invenio_rdm_records/config.py]: Are these vocabularies somehow?

File: invenio_rdm_records/cli.py (hunk 1 above)
Comment (ppanero) [reply invenio_rdm_records/cli.py]: I was just trying not to collide with anything non-existent :) but having rdm there already ensures that. Chaging.

File: invenio_rdm_records/theme/__init__.py (hunk 2 above)
Comment (ppanero) [reply invenio_rdm_records/theme/__init__.py]: True, removed all.

File: invenio_rdm_records/theme/templates/invenio_rdm_records/macros/files.html (hunk 4 above)
Comment (ppanero) [reply invenio_rdm_records/theme/templates/invenio_rdm_records/macros/files.html]: That one was already in app-rdm from previous releases, so I just moved it here. It's in my todo list to analyze which macros exist and if some can be moved to other repos :)

File: invenio_rdm_records/config.py (hunk 5 above)
Comment (ppanero) [reply invenio_rdm_records/config.py]: As talked IRL. Yes, but to be decided how we handle them in the future :)

File: invenio_rdm_records/theme/__init__.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+# Copyright (C) 2020 Northwestern University.
+#
+# Invenio RDM Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio RDM Records."""
+
+from ..version import __version__
+
+__all__ = ('__version__', )
```
Comment (fenekku) [inline invenio_rdm_records/theme/__init__.py]: why is this file necessary?

File: invenio_rdm_records/theme/assets/scss/invenio_rdm_records/theme.scss [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+ *   Copyright (C) 2020 Northwestern University.
+ *
+ * Invenio RDM Records is free software; you can redistribute it and/or modify
+ * it under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+@import "variables";
+
+// Preview
+.iiif-preview-img {
```
Comment (fenekku) [inline invenio_rdm_records/theme/assets/scss/invenio_rdm_records/theme.scss]: minor: Could move

```css
.preview-iframe {
  border-style: none;
}
```

from invenio-app-rdm here too.

File: invenio_rdm_records/theme/assets/scss/invenio_rdm_records/theme.scss [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+.thin-line {
+    margin: 0px !important;
+}
+
+.send-btn {
+    margin-bottom: 10px;
+    padding: 2px 6px;
+}
```
Comment (fenekku) [inline invenio_rdm_records/theme/assets/scss/invenio_rdm_records/theme.scss]: We might need more padding in the bottom to compensate for the whitespace at the top of the button:

image

File: invenio_rdm_records/theme/assets/scss/invenio_rdm_records/variables.scss [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+ *
+ * Invenio RDM Records is free software; you can redistribute it and/or modify
+ * it under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+
+$brand-primary: #0377cd;
+
+// This is nowhere in Zenodo nor Invenio (?)
+$font-size-base: 14px;
```
Comment (fenekku) [inline invenio_rdm_records/theme/assets/scss/invenio_rdm_records/variables.scss]: just shows up in bootstrap for me...

File: invenio_rdm_records/theme/static/images/orcid.svg [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+<path fill="#A6CE39" d="M255.999,128.008C255.999,198.697,198.69,256,127.997,256C57.306,256,0,198.697,0,128.008
+	S57.306,0,127.997,0C198.69,0,255.999,57.318,255.999,128.008z"/>
+<g>
+	<path fill="#FFFFFF" d="M86.337,186.215H70.943V79.102h15.394v48.387V186.215z"/>
+	<path fill="#FFFFFF" d="M108.879,79.102h41.624c39.578,0,57,28.271,57,53.559c0,27.486-21.506,53.555-56.844,53.555h-41.78V79.102z
+		 M124.283,172.379h24.495c34.859,0,42.867-26.531,42.867-39.719c0-21.527-13.66-39.746-43.652-39.746h-23.71V172.379z"/>
+	<path fill="#FFFFFF" d="M88.728,56.811c0,5.541-4.538,10.096-10.086,10.096c-5.555,0-10.094-4.555-10.094-10.096
+		c0-5.555,4.539-10.102,10.094-10.102C84.189,46.709,88.728,51.256,88.728,56.811z"/>
+</g>
+</svg>
```
Comment (fenekku) [inline invenio_rdm_records/theme/static/images/orcid.svg]: Both png and svg are needed?

File: invenio_rdm_records/theme/templates/invenio_rdm_records/coming_soon_page.html [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    <br>
+    <br>
+    <br>
+    <br>
+    <br>
+    <br>
+    <br>
+    <br>
+    <br>
+    <br>
```
Comment (fenekku) [inline invenio_rdm_records/theme/templates/invenio_rdm_records/coming_soon_page.html]: Made an issue for "sticky footer": https://github.com/inveniosoftware/invenio-app-rdm/issues/81. I've done it in the past for our instance.

File: invenio_rdm_records/theme/templates/invenio_rdm_records/details/contributors.html [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+{%- for group in record.contributors|groupby('role')%}
+<h5><b>{{group.grouper|contributortype_title}}(s)</b></h5>
+
+{%- for creator in group.list %}
+{%- if creator.identifiers[0].scheme == "Orcid" %}
+<a href="{{creator.identifiers[0].identifier|pid_url('orcid')}}">
+    <img class="inline-orcid" src="{{ url_for('static', filename='images/orcid.png')}}" />
+</a>
+{%- endif %}
+<span class="text-muted" {% if creator.affiliations and creator.affiliations[0] %}data-toggle="tooltip" title="{{creator.affiliations[0].name}}"{% endif %}>{{creator.name}}</span>{% if not loop.last %}; {% endif %}
```
Comment (fenekku) [inline invenio_rdm_records/theme/templates/invenio_rdm_records/details/contributors.html]: future/custom: could eventually get away with no `;` and always have an icon (even if generic) next to a creator

File: invenio_rdm_records/theme/templates/invenio_rdm_records/details/contributors.html [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+    Copyright (C) 2020 Northwestern University.
+  
+    Invenio RDM Records is free software; you can redistribute it and/or modify
+    it under the terms of the MIT License; see LICENSE file for more details.
+#}
+
+{%- for group in record.contributors|groupby('role')%}
+<h5><b>{{group.grouper|contributortype_title}}(s)</b></h5>
+
+{%- for creator in group.list %}
```
Comment (fenekku) [inline invenio_rdm_records/theme/templates/invenio_rdm_records/details/contributors.html]: `creator` -> `contributor` or we can use macro to reuse across contributor and creators.

File: invenio_rdm_records/theme/templates/invenio_rdm_records/details/doi.html [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+  Copyright (C) 2020 Northwestern University.
+
+  Invenio RDM Records is free software; you can redistribute it and/or modify
+  it under the terms of the MIT License; see LICENSE file for more details.
+#}
+
+{%- set id_doi = record.identifiers|doi_identifier %}
+<dt>{{_('DOI')}}:</dt>
+<dd>
+  <span class="get-badge" data-toggle="tooltip" data-placement="bottom"
```
Comment (fenekku) [inline invenio_rdm_records/theme/templates/invenio_rdm_records/details/doi.html]: minor: I would never have known this was possible without looking at the code! Neat. Could we add css to make cursor a pointer: `cursor: pointer;` when hovering over the badge?

File: invenio_rdm_records/theme/templates/invenio_rdm_records/details/side_bar.html [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+          <div class="col-sm-4">
+              <i class="fa fa-pencil " data-toggle="tooltip" tooltip="" data-placement="top" title="" data-original-title="Total citation."></i> citations
+          </div>
+      </div>
+
+    <!-- Collapsed details -->
+    <div id="toggle-stats" class="row">
+      <div class="col-sm-12">
+        <a style="cursor: pointer;" class="panel-toggle" data-toggle="collapse" data-target="#collapse-stats" aria-expanded="true" aria-controls="collapse-stats">
+          See more details...
```
Comment (fenekku) [inline invenio_rdm_records/theme/templates/invenio_rdm_records/details/side_bar.html]: should say "See less details" when unrolled. This can be shelved for when we integrate invenio-stats.

File: invenio_rdm_records/theme/templates/invenio_rdm_records/record_landing_page.html [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+        <span class="label label-success titlecase">
+        {% endif %}
+        {{ record.access_right }} Access
+        </span>
+      </div>
+
+      <h1>{{ record.titles[0].title }}</h1>
+      <p>{%- include "invenio_rdm_records/details/creators.html" %}</p>
+      {%- if record.contributors %}
+      <p>{%- include "invenio_rdm_records/details/contributors.html" %}</p>
```
Comment (fenekku) [inline invenio_rdm_records/theme/templates/invenio_rdm_records/record_landing_page.html]: I think a `<h3>Contributors</h3>` here would help separate creators from contributors (especially since right now the "Rights holder(s)" shows up and it's surprising.

File: invenio_rdm_records/theme/templates/invenio_rdm_records/record_landing_page.html [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+
+    {# Side bar #}
+    <div class="col-sm-4 col-md-4 col-right">
+      <div class="well">
+        {%- include "invenio_rdm_records/details/side_bar.html" %}
+      </div>
+    </div>
+  </div>
+</div>
+{{ webpack['invenio-rdm-records-js.js'] }}
```
Comment (fenekku) [inline invenio_rdm_records/theme/templates/invenio_rdm_records/record_landing_page.html]: Why is this not in the javascript block?

File: setup.py [hunk 18]
```diff
@@ -32,6 +32,9 @@
 invenio_search_version = '>=1.2.0,<1.3.0'
 
 extras_require = {
+    'badges': [
+        'CairoSVG>=1.0.20',
+    ],
```
Comment (fenekku) [inline setup.py]: curious: what makes the cut for an `extras_require` and what doesn't?

File: examples/app.py [hunk 19]
```diff
@@ -42,13 +42,11 @@
 SPHINX-END
 """
 
-from __future__ import absolute_import, print_function
```
[the remark this answers is not in the document]
Comment (ppanero) [reply examples/app.py]: :fire: :fire_engine:

File: invenio_rdm_records/theme/templates/invenio_rdm_records/coming_soon_page.html (hunk 11 above)
Comment (ppanero) [reply invenio_rdm_records/theme/templates/invenio_rdm_records/coming_soon_page.html]: It's true, tho also the coming soon is temporary. But in any case it would be nice to have the footer sticky at the bottom in case of few results in search or so :100:

File: invenio_rdm_records/theme/assets/scss/invenio_rdm_records/theme.scss (hunk 7 above)
Comment (ppanero) [reply invenio_rdm_records/theme/assets/scss/invenio_rdm_records/theme.scss]: True! Thanks for spotting!

File: invenio_rdm_records/theme/templates/invenio_rdm_records/coming_soon_page.html (hunk 11 above)
Comment (zzacharo) [reply invenio_rdm_records/theme/templates/invenio_rdm_records/coming_soon_page.html]: @fenekku @ppanero perfect because I was about to make my footer sticky :) So I will leave it for another day or should I do it?

File: invenio_rdm_records/theme/assets/scss/invenio_rdm_records/theme.scss (hunk 8 above)
Comment (ppanero) [reply invenio_rdm_records/theme/assets/scss/invenio_rdm_records/theme.scss]: It displays correctly to me, the way it is padding should be ok both up and bottom. Which would work for you? We can test different ones :)

image

File: invenio_rdm_records/theme/assets/scss/invenio_rdm_records/variables.scss (hunk 9 above)
Comment (ppanero) [reply invenio_rdm_records/theme/assets/scss/invenio_rdm_records/variables.scss]: Yup, in Zenodo it's a variable but not defining it was failing to me (even when importing bootstrap). Since we are moving to Semantic-UI I'll just leave it as is for now if it's okay for you

File: invenio_rdm_records/theme/static/images/orcid.svg (hunk 10 above)
Comment (ppanero) [reply invenio_rdm_records/theme/static/images/orcid.svg]: You are right. For now we are only using the png, which is less scalable. I change the usage to svg and removed the png file :).

File: invenio_rdm_records/theme/templates/invenio_rdm_records/details/contributors.html (hunk 13 above)
Comment (ppanero) [reply invenio_rdm_records/theme/templates/invenio_rdm_records/details/contributors.html]: I didnt create a macro cuz it felt a bit overkill, its true that is repeated code. But also I am not sure how the UI will evolve. I guess we will be creating macros in the next iterations when we have more and more UI :)

changed to contributor :)

File: invenio_rdm_records/theme/templates/invenio_rdm_records/details/doi.html (hunk 14 above)
Comment (ppanero) [reply invenio_rdm_records/theme/templates/invenio_rdm_records/details/doi.html]: Yup, done :)

File: invenio_rdm_records/theme/templates/invenio_rdm_records/record_landing_page.html (hunk 17 above)
Comment (ppanero) [reply invenio_rdm_records/theme/templates/invenio_rdm_records/record_landing_page.html]: Cuz I missed it, thanks! :)

File: invenio_rdm_records/theme/templates/invenio_rdm_records/details/side_bar.html (hunk 15 above)
Comment (ppanero) [reply invenio_rdm_records/theme/templates/invenio_rdm_records/details/side_bar.html]: Done :) tho to be re-checked when integrating invenio-stats :)

File: invenio_rdm_records/theme/templates/invenio_rdm_records/details/contributors.html (hunk 12 above)
Comment (ppanero) [reply invenio_rdm_records/theme/templates/invenio_rdm_records/details/contributors.html]: https://github.com/inveniosoftware/invenio-rdm-records/issues/61

File: invenio_rdm_records/theme/templates/invenio_rdm_records/record_landing_page.html (hunk 16 above)
Comment (ppanero) [reply invenio_rdm_records/theme/templates/invenio_rdm_records/record_landing_page.html]: Yup, nice point :100:

File: invenio_rdm_records/theme/templates/invenio_rdm_records/coming_soon_page.html (hunk 11 above)
Comment (ppanero) [reply invenio_rdm_records/theme/templates/invenio_rdm_records/coming_soon_page.html]: Leave it for later, unless it takes you just a few seconds :)

File: setup.py (hunk 18 above)
Comment (ppanero) [reply setup.py]: AFAIK usually thinks that might be optional or choice (e.g. ES versions). But it's true in this case it fails to build otherwise. I was just thinking that not all UIs might want the badges... but it's just an assumption lets see what happens in reality

Moved to main install requirements

File: invenio_rdm_records/theme/__init__.py (hunk 6 above)
Comment (ppanero) [reply invenio_rdm_records/theme/__init__.py]: I think is not. It's a port from cookiecutter-invenio-instance/records module. I think the file is, the import of version is not: https://github.com/inveniosoftware/invenio-rdm-records/issues/63

Comment (fenekku) [reply invenio_rdm_records/theme/__init__.py]: > the import of version is not

Haha that's what I meant. I am fine with the presence of __init__.py :)

File: invenio_rdm_records/theme/assets/scss/invenio_rdm_records/theme.scss (hunk 8 above)
Comment (fenekku) [reply invenio_rdm_records/theme/assets/scss/invenio_rdm_records/theme.scss]: the one you posted looks better than mine. I was on Firefox 73

File: invenio_rdm_records/theme/assets/scss/invenio_rdm_records/variables.scss (hunk 9 above)
Comment (fenekku) [reply invenio_rdm_records/theme/assets/scss/invenio_rdm_records/variables.scss]: Sure.
